libvmi/events.c

```diff
@@ -1060,10 +1061,31 @@ vmi_stop_single_step_vcpu(
         return VMI_FAILURE;
```

Comment:
You should add a safety check that checks if this was called from an event callback. For that use-case there is an event_response_t flag that should be used.

Reply:
Done